core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2114,6 +2114,16 @@ public Label getLabel(String expr) {
         }
     }
 
+    /**
+     * Returns the label atom of the given name, only if it already exists.
+     * @return non-null if the label atom already exists.
+     */
+    public @Nullable LabelAtom tryGetLabelAtom(@CheckForNull String name) {
+        if (name == null)  return null;
+        return (LabelAtom) labels.get(name);
```

Comment:
Will this not throw a CCE if it is some other kind of `Label`?

AFAICT https://github.com/jenkinsci/jenkins/blob/a17f0f63c9fac78d3afc4f7864aa0399d6f1c7b4/core/src/main/java/jenkins/model/Jenkins.java#L2100-L2106 is also unsafe in general. I must be missing something.

```diff
diff --git test/src/test/java/hudson/model/labels/LabelExpressionTest.java test/src/test/java/hudson/model/labels/LabelExpressionTest.java
index 4716daa2ba..049a0a7f96 100644
--- test/src/test/java/hudson/model/labels/LabelExpressionTest.java
+++ test/src/test/java/hudson/model/labels/LabelExpressionTest.java
@@ -241,6 +241,12 @@ public class LabelExpressionTest {
         assertEquals("label1||label2", l.getName());
     }
 
+    @Test
+    public void wtf() {
+        j.jenkins.getLabel("xxx && yyy");
+        j.jenkins.getLabelAtom("xxx && yyy");
+    }
+
     /**
      * The name should have parenthesis at the right place to preserve the tree structure.
      */
```

⇒

```
java.lang.ClassCastException: class hudson.model.labels.LabelExpression$And cannot be cast to class hudson.model.labels.LabelAtom (hudson.model.labels.LabelExpression$And and hudson.model.labels.LabelAtom are in unnamed module of loader 'app')
	at jenkins.model.Jenkins.getLabelAtom(Jenkins.java:2106)
	at hudson.model.labels.LabelExpressionTest.wtf(LabelExpressionTest.java:247)
```

I guess `getLabelAtom` (and `LabelAtom.get`) is just called from contexts where an atom is actually expected; whereas this new method is called from `Label.parse` and would routinely be given a label expression?